fact/src/lib.rs

```diff
@@ -31,7 +31,12 @@ use event::Event;
 use pre_flight::pre_flight;
 
 pub async fn run(config: FactConfig) -> anyhow::Result<()> {
-    pre_flight().context("Pre-flight checks failed")?;
+    if !config.skip_preflight {
+        debug!("Performing pre-flight checks");
+        pre_flight().context("Pre-flight checks failed")?;
+    } else {
+        debug!("Skipping pre-flight checks");
+    }
```

Comment:
Wonder if it'd be cleaner to move the check inside `pre_flight`, pass the boolean in and do an early return. Here it would look like this:

```rust
    pre_flight(config.skip_preflight).context("Pre-flight checks failed")?;
```

And on the `pre_flight` it'd be:
```rust
pub fn pre_flight(skip: bool) -> anyhow::Result<()> {
    if skip {
        debug!("Skipping pre-flight checks");
        return Ok(());
    }
    
    debug!("Performing pre-flight checks");
    have_bpf_lsm()
}
```

I think having a smaller `run` method is a bit cleaner, (#3 will get rid of a lot of the logic in here), but ultimately it is a matter of personal preference, so I'll let you decide if you want to make this change.

Reply:
It feels strange to have an argument function to not invoke the function, so I would prefer to keep it outside. If the `run` method will become bloated, we could introduce an indirection level to wrap this into a simple function and hide skip logic inside, then from it call the actual pre-flight check. Makes sense?